client/components/Draft.jsx

```diff
@@ -1,42 +1,97 @@
-import React, { useState } from "react";
+import React, { createRef, useState, useEffect } from "react";
+import { DraftAddImage } from "../components";
+import { Row, Col } from "reactstrap";
+import { EditorState, convertToRaw, convertFromRaw } from "draft-js";
+import { Editor, createEditorState } from "medium-draft";
 
-import { addNewBlock } from "medium-draft";
+import { saveDescription, getDescription } from "../utils/apiWrapper";
+
+import "../public/styles/medium-draft.scss";
+import debounce from "lodash/debounce";
 
 export default function Draft(props) {
-  const [input, setInput] = useState(null);
-
-  const handleClick = () => input.click();
-
-  const handleChange = e => {
-    const { close, getEditorState, setEditorState } = props;
-    const file = e.target.files[0];
-    if (file.type.indexOf("image/") === 0) {
-      let reader = new FileReader();
-      reader.readAsDataURL(file);
-      reader.onload = () => {
-        const src = reader.result;
-        const newState = addNewBlock(getEditorState(), "atomic:image", { src });
-        setEditorState(newState);
-      };
+  const [editorState, setEditorState] = useState(createEditorState());
+  const [unsaved, setUnsaved] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [prevContent, setPrevContent] = useState(null);
+
+  const refsEditor = createRef();
+
+  const saveInterval = 1000;
+  const debounceSave = React.useCallback(
+    debounce(json => {
+      const { id, phaseName, stageName } = props;
```

Comment:
can destructure outside of individual functions (at the top level fo the component) so that you dont need to redo every time

Reply:
What do you mean? Make a separate function for the `json => { ... }`?

Reply:
oh like you could do `const { id, phaseName, stageName } = props;` at the same level as your other functions so that all your functions would have access to it but thats a super super nitpicky thing so dw